### PR TITLE
feat: RENOVATE_X_HARD_EXIT

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds an experimental env `RENOVATE_X_HARD_EXIT` which performs a `process.exit()` if set.

## Context:

Workaround for ECR problems in #8660.

Not sure where/how we should document this? We have some other similar env variables too, but this is the first time I am using the `RENOVATE_X_` naming convention. The idea is that these experimental options are not otherwise exposed to config or polluting our docs, and hopefully each will go away over time or be promoted to a first-class option if necessary. However should we add a new "experimental environment options" section or page in our docs?

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/renovate.ts
Comment (viceice) [inline lib/renovate.ts:12]:
```suggestion
  if (process.env.RENOVATE_EXPERIMENTAL_HARD_EXIT) {
```

Because we're already using `RENOVATE_EXPERIMENTAL_NO_MAVEN_POM_CHECK` ?

**Review by viceice — COMMENTED** [review]
LGTM, just one small thing.

**Comment by viceice** [comment]
Yeah, maybe adding an experimental docs page for this, and warn that those can changed any time with any release

File: lib/renovate.ts
Comment (rarkins) [reply lib/renovate.ts:12]: I think it's pretty long, and I was also basing it off the idea of `X-` fields/headers in other areas of software. I'd prefer to standardize on `RENOVATE_X_` from now on

**Review by viceice — DISMISSED** [review]
So we should change the other `RENOVATE_EXPERIMENTAL_**` too (with migration and warning)

**Comment by rarkins** [comment]
I'm going to document the others in a commit to master first and then add this one to the docs once I've merged it into this branch. Won't rename right away.

**Comment by rarkins** [comment]
Blocked by #9156

File: docs/usage/self-hosted-experimental.md
Comment (HonkingGoose) [inline docs/usage/self-hosted-experimental.md:45]: Most of the time you need to use a specific number/string/boolean to get things configured properly.

Are you sure literally any value will work to enable `RENOVATE_X_HARD_EXIT`?
I think it's clearer to say: "Set to `x`", where `x` is a specific value.

**Review by viceice — APPROVED** [review]

File: docs/usage/self-hosted-experimental.md
Comment (rarkins) [reply docs/usage/self-hosted-experimental.md:45]: Any value will do - true, yes, y, no, never, HonkingGoose. Basically it just needs to be "set"

Comment (HonkingGoose) [reply docs/usage/self-hosted-experimental.md:45]: Ah okay, we can keep the text as-is then. 😄

**Pull request merged**
